pkg/watch/health.go

```diff
@@ -0,0 +1,257 @@
+		p.lastCheck, err = writeToConsul(health, store)
+		p.lastStatus = health.Status
+		if err != nil {
+			p.logger.WithField("err", err).Warningln("failed to write health data to consul")
+		}
+	}
+}
+
+func updateHealthMonitors(store kp.Store, watchedPods []*PodWatch, node string, logger *logging.Logger) []*PodWatch {
+	path := kp.RealityPath(node)
+	reality, _, err := store.ListPods(path)
+	if err != nil {
+		logger.WithField("inner_err", err).Warningln("failed to get pods from reality store")
+		return nil
+	}
+
+	// update list of pods to be monitored
+	watchedPods = updatePods(watchedPods, reality, logger)
+	for _, pod := range watchedPods {
+		if pod.logger == nil {
+			pod.logger = logger
+		}
+		if pod.hasMonitor == false {
+			go pod.MonitorHealth(node, store, pod.shutdownCh)
+		}
+	}
+
+	return watchedPods
+}
+
+func resultFromCheck(resp *http.Response) (health.Result, error) {
+	var err error
+	res := health.Result{}
+	if resp == nil {
+		return res, nil
+	} else if resp.StatusCode == 200 {
+		res.Status = health.Passing
+	} else if resp.StatusCode > 200 && resp.StatusCode < 300 {
+		res.Status = health.Warning
+		res.Output, err = getBody(resp)
+	} else {
+		res.Status = health.Critical
+		res.Output, err = getBody(resp)
+	}
+	return res, err
+}
+
+func getBody(resp *http.Response) (string, error) {
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	return string(body), nil
+}
+
+// once we get health data we need to make a put request
+// to consul to put the data in the KV Store
+func writeToConsul(res health.Result, store kp.Store) (time.Time, error) {
+	timeOfPut, _, err := store.PutHealth(resToMap(res))
+	return timeOfPut, err
+}
+
+func resToMap(res health.Result) map[string]string {
+	m := make(map[string]string)
+	m["service"] = res.Service
+	m["node"] = res.Node
+	m["id"] = res.ID
+	m["status"] = string(res.Status)
+	m["output"] = res.Output
+
+	return m
+}
+
+func (p *PodWatch) updateNeeded(res health.Result, ttl int) bool {
+	// if status has changed indicate that consul needs to be updated
+	if p.lastStatus != res.Status {
+		return true
+	}
+	// if more than TTL / 4 seconds have elapsed since previous check
+	// indicate that consul needs to be updated
+	if time.Since(p.lastCheck) > time.Duration(ttl/4)*time.Second {
+		return true
+	}
+
+	return false
+}
+
+// compares services being monitored with services that
+// need to be monitored.
+func updatePods(current []*PodWatch, reality []kp.ManifestResult, logger *logging.Logger) []*PodWatch {
+	newCurrent := []*PodWatch{}
+	// for pod in current if pod not in reality: kill
+	for _, pod := range current {
+		inReality := false
+		for _, man := range reality {
+			if man.Manifest.Id == pod.manifest.Manifest.Id {
+				inReality = true
+				break
+			}
+		}
+
+		// if this podwatch is not in the reality store kill its go routine
+		// else add this podwatch to newCurrent
+		if inReality == false {
```

Comment:
Nit: Awkward test `if !thing {a} else {b}`. Invert so that it's `if thing {b} else {a}`